blockchain/message_pool/src/msgpool.rs

```diff
@@ -467,13 +445,14 @@ where
     /// Load local messages into pending. As of  right now messages are not deleted from self's
     /// local_message field, possibly implement this in the future?
     pub async fn load_local(&mut self) -> Result<(), Error> {
-        for (key, value) in self.local_msgs.clone() {
-            self.add(&value).await.unwrap_or_else(|err| {
+        let msg_vec = self.local_msgs.clone();
+        for msg in msg_vec {
+            self.add(&msg).await.unwrap_or_else(|err| {
                 if err == Error::SequenceTooLow {
                     warn!("error adding message: {:?}", err);
-                    self.local_msgs.remove(&key);
+                    self.local_msgs.retain(|smsg| *smsg != msg);
```

Comment:
why did you switch from using a hash map for local_msgs? Removing from the vec and having to do these checks is not very efficient

Reply:
I was thinking that using a vec and its inefficiencies of this was better than having to clone the hashmap

Reply:
you don't need to clone the hashmap, and yeah this ends up being not very efficient for it to be a vec instead

Reply:
I don't see how you don't have to clone the hashmap, Reason being is self.add needs a reference to a signed message, and if you don't clone the hm, borrow checker get mad.

Reply:
Hashmaps have a function, `drain`, which would probably have solved your issue

Reply:
I was thinking about using drain, the only issue with it was that after all the kv pairs are drained from local_msgs, a temp variable would be needed to store the values that need to be reinserted into local_msgs. What I was thinking of doing was using a Box<Vec<SignedMessage>> for local_msgs, and doing something like the following:

```
let mut local = Vec::new();

while let Some(msg) = self.local_msgs.pop() {
    self.add(&msg).await.unwrap_or_else(|err| {
        if err == Error::SequenceTooLow {
            warn!("error adding message: {:?}", err);
        } else {
            local.push(msg);
        }
    })
}
*self.local_msgs = local;
```